### PR TITLE
760 add dropdown for task status mobile

### PR DESCRIPTION
**Pull request by poornima-alugubelly**
### Issue:
https://github.com/Real-Dev-Squad/website-status/issues/760

### Description:
Currently the task status tabs look cluttered in mobile.Added dropdown for task status in mobile so it looks cleaner.
Created Select component for the same and added tests as well

### Dev Tested:
- Yes
Test coverage:
image

-------------------------------

image

### Images/video of the change:
image

### CODE REVIEW

File: src/components/Tabs/index.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ type TabsProps = {
     onSelect: (tab: Tab) => void;
     activeTab: Tab;
 };
-function changeName(name: string) {
+export function changeName(name: string) {
```
Comment (bhtibrewal) [inline src/components/Tabs/index.tsx:10]: Suggestion: if this function is being used in other components, you can move it to a util file

File: __tests__/Unit/Components/Select/Select.test.tsx [hunk 2]
```diff
@@ -0,0 +1,182 @@
+import React from 'react';
+import { render, fireEvent, screen } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect'; // For additional DOM matchers
```
Comment (bhtibrewal) [inline __tests__/Unit/Components/Select/Select.test.tsx:3]: this is not needed it is already present in jest.setup file

File: __tests__/Unit/Components/Select/Select.test.tsx [hunk 3]
```diff
@@ -0,0 +1,182 @@
+import React from 'react';
```
Comment (bhtibrewal) [inline __tests__/Unit/Components/Select/Select.test.tsx:1]: need of importing react in the test file?

File: src/components/tasks/TasksContent.tsx [hunk 4]
```diff
@@ -109,18 +111,38 @@ export const TasksContent = ({ dev }: { dev: boolean }) => {
...
+                    value={{
+                        label: changeName(selectedTab),
+                        value: selectedTab,
+                    }}
+                    onChange={(o) => {
+                        if (o) {
+                            console.log(o);
+                            onSelect(o.value as Tab);
+                        }
+                    }}
```
Comment (bhtibrewal) [inline src/components/tasks/TasksContent.tsx:135]: can we name this something more meaningful other than `o`
also please remove the console.log

**Review by bhtibrewal — COMMENTED** [review]
left some comments, please have a look at them, also please write tests for `src/components/tasks/TasksContent.tsx`
also can you please add coverage report image in the PR description

File: src/components/Tabs/index.tsx (hunk 1 above)
Comment (poornima-alugubelly) [reply src/components/Tabs/index.tsx]: resolved

File: __tests__/Unit/Components/Select/Select.test.tsx (hunk 2 above)
Comment (poornima-alugubelly) [reply __tests__/Unit/Components/Select/Select.test.tsx]: resolved

File: __tests__/Unit/Components/Select/Select.test.tsx (hunk 3 above)
Comment (poornima-alugubelly) [reply __tests__/Unit/Components/Select/Select.test.tsx]: resolved

File: src/components/tasks/TasksContent.tsx (hunk 4 above)
Comment (poornima-alugubelly) [reply src/components/tasks/TasksContent.tsx]: resolved

**Comment by poornima-alugubelly** [comment]
> left some comments, please have a look at them, also please update tests for `src/components/tasks/TasksContent.tsx`

added tests and test coverage is in PR description

File: src/components/Select/index.tsx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+                        className={`${
+                            index === highlightedIndex ? styles.highlighted : ''
+                        } ${styles.option}`}
+                    >
+                        {option.label}
+                    </li>
+                ))}
+            </ul>
+        </div>
+    );
```
Comment (bharati-21) [inline src/components/Select/index.tsx:99]: Great work!

File: __tests__/Unit/Components/Tasks/TasksContent.test.tsx [hunk 6]
```diff
@@ -165,4 +194,70 @@ describe('tasks content', () => {
...
+
+        fireEvent.keyDown(selectContainer, {
+            key: 'Enter',
+            code: 'Enter',
+        });
+
+        expect(mockPushFunction).toBeCalledTimes(1);
+        expect(mockPushFunction).toBeCalledWith({
+            query: {
+                section: 'assigned',
```
Comment (bharati-21) [inline __tests__/Unit/Components/Tasks/TasksContent.test.tsx:259]: If the order of tabs changes, this wouldn't still be 'assigned' right? Can we please use the `Tabs` enum to select the 2nd option?

File: __tests__/Unit/Components/Select/Select.test.tsx [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+        const value = sampleOptions[0];
+        render(
+            <Select
+                value={value}
+                onChange={() => onSelectMock()}
+                options={sampleOptions}
+            />
+        );
+
+        expect(screen?.queryByTestId('selected-option')).toBeInTheDocument();
```
Comment (bharati-21) [inline __tests__/Unit/Components/Select/Select.test.tsx:22]: Why optional chaining?

**Comment by bharati-21** [comment]
Great work with testing! Next time please do attach a short video to show the changes.

File: __tests__/Unit/Components/Tasks/TasksContent.test.tsx (hunk 6 above)
Comment (poornima-alugubelly) [reply __tests__/Unit/Components/Tasks/TasksContent.test.tsx]: Yes you're right made it dynamic now by using the tabs enum

File: __tests__/Unit/Components/Select/Select.test.tsx (hunk 7 above)
Comment (poornima-alugubelly) [reply __tests__/Unit/Components/Select/Select.test.tsx]: Not needed, removed

**Review by bhtibrewal — APPROVED** [review]

**Review by bharati-21 — APPROVED** [review]
LGTM

**Review by harshith-venkatesh — APPROVED** [review]

**Pull request merged**